boards/nxp/mr_canhubk3/mr_canhubk3.dts

```diff
@@ -616,6 +617,33 @@
 	};
 };
 
+&flexio0 {
+	status = "okay";
+
+	flexio0_pwm: flexio0_pwm {
+		pinctrl-0 = <&flexio0_pwm_default>;
+		pinctrl-names = "default";
+		status = "okay";
+
+		pwm_0 {
+			pin-id = <19>;
+			period = <400>;
+			duty-cycle = <200>;
+			prescaler = <0>;
+			polarity = "active-high";
+		};
```

Comment:
This is not typically how we configure PWM outputs in devicetree? Configuration is done from the PWM client node, using a `*pwms = <&flexio0_pwm ...>` property.

Reply:
@henrikbrixandersen 
Shall I remove only polarity or both polarity and period from the device tree properties (as both polarity and period are a part of pwm-cells)

@manuargue .. 
There is no constraint to set the polarity at the init time in flexio PWM mode.
But if I need to remove period as well from the device tree then I will get a low flat pwm line initially and will definitely need a pwm client entry in my pwm app overlay like this -

 pwm_<app>_0 {
                compatible = "test-pwm-<app>";
                pwms = <&flexio0_pwm 0 0 PWM_POLARITY_NORMAL>,
                        <&flexio0_pwm 1 0 PWM_POLARITY_NORMAL>;
        };

and then use macros like - DT_PWMS_PERIOD_BY_IDX, DT_PWMS_FLAGS_BY_IDX in the driver to get the period, polarity at init time

Reply:
@manuargue .. So is it ok if I don't make any overlay entries in a pwm app (samples/test) for this driver.. ?

Reply:
> Shall I remove only polarity or both polarity and period from the device tree properties (as both polarity and period are a part of pwm-cells)

I would remove both polarity, duty-cycle and period.

Reply:
@henrikbrixandersen .. I understand that pwm-cells has period and flags (polarity and more), so there is no reason to have duplicate properties but why duty cycle needs to be removed ?

Reply:
It is up to the PWM client to specify the duty cycle as well as the other properties. 

I suggest taking a look at how all the other PWM drivers operate.

Reply:
> > So is it ok if I don't make any overlay entries for pwm client nodes in a pwm app (samples/test) for this driver.. ?
> 
> Ideally we should have something to build or test this driver.

@manuargue .. 
tests/drivers/pwm/pwm_api can be used to operate this driver as flexio0_pwm node is pwm-1 alias
But for depicting pwms client node for applications -
I cannot use - tests/drivers/pwm/pwm_loopback as flexio doesn't have a pwm input mode like emios SAIC
So, shall I create an overlay in boards/nxp/mr_canhubk3 or do you suggest any other existing application ?

Reply:
So I have not added anything on the application side, but have removed period, duty_cycle and polarity from the device tree and bindings as suggested by @henrikbrixandersen
But have hardcoded these values for init time in the driver.

Reply:
By default, no PWM signal should be produced until `pwm_set_cycles_t` for the given driver is called. You cannot have a default PWM period and duty-cycle upon driver initialisation.

Reply:
Removed default PWM period and duty-cycle upon driver initialization..